<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.1.189">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Somite periodicity notebook - 3&nbsp; F0 homozygosity</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<link href="./04-F1-heterozygosity.html" rel="next">
<link href="./02-phenotypes.html" rel="prev">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>


</head>

<body class="nav-sidebar floating">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
    <div class="container-fluid d-flex justify-content-between">
      <h1 class="quarto-secondary-nav-title"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">F<sub>0</sub> homozygosity</span></h1>
      <button type="button" class="quarto-btn-toggle btn" aria-label="Show secondary navigation">
        <i class="bi bi-chevron-right"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="./">Somite periodicity notebook</a> 
        <div class="sidebar-tools-main">
    <a href="https://github.com/brettellebi/somites/tree/master/book/" title="Source Code" class="sidebar-tool px-1"><i class="bi bi-github"></i></a>
</div>
    </div>
      </div>
      <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
      </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">About</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./01-background.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Background</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./02-phenotypes.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Phenotypes of interest</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./03-F0-homozygosity.html" class="sidebar-item-text sidebar-link active"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">F<sub>0</sub> homozygosity</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./04-F1-heterozygosity.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">F<sub>1</sub> heterozygosity</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./07-references.html" class="sidebar-item-text sidebar-link">References</a>
  </div>
</li>
    </ul>
    </div>
</nav>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#f0-coverage" id="toc-f0-coverage" class="nav-link active" data-scroll-target="#f0-coverage"><span class="toc-section-number">3.1</span>  F<sub>0</sub> coverage</a></li>
  <li><a href="#f0-homozygosity" id="toc-f0-homozygosity" class="nav-link" data-scroll-target="#f0-homozygosity"><span class="toc-section-number">3.2</span>  F<sub>0</sub> homozygosity</a></li>
  </ul>
<div class="toc-actions"><div><i class="bi bi-github"></i></div><div class="action-links"><p><a href="https://github.com/brettellebi/somites/edit/master/book/03-F0-homozygosity.qmd" class="toc-action">Edit this page</a></p></div></div></nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title d-none d-lg-block"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">F<sub>0</sub> homozygosity</span></h1>
</div>



<div class="quarto-title-meta">

    
    
  </div>
  

</header>

<section id="f0-coverage" class="level2" data-number="3.1">
<h2 data-number="3.1" class="anchored" data-anchor-id="f0-coverage"><span class="header-section-number">3.1</span> F<sub>0</sub> coverage</h2>
<p>Our collaborators extracted DNA from the F<sub>0</sub>, F<sub>1</sub>, and F<sub>2</sub>, and sequenced the F<sub>0</sub> and F<sub>1</sub> samples with the Illumina platform at high coverage (~26x for <em>Cab</em> and ~29x for <em>Kaga</em>), as measured by <em>SAMtools</em> <span class="citation" data-cites="danecekTwelveYearsSAMtools2021">(<a href="07-references.html#ref-danecekTwelveYearsSAMtools2021" role="doc-biblioref">Danecek et al. 2021</a>)</span>. <strong><a href="#fig-F0-coverage">Figure&nbsp;<span>3.1</span></a></strong> sets out the mean sequencing depth within each chromosome and across the whole genome for the <em>Cab</em> and <em>Kaga</em> F<sub>0</sub> samples. Our collaborators then sequenced the F<sub>2</sub> samples at low coverage (~1x), which would be sufficient to map their genotypes back to the genotypes of their parental strains (see section @ref(somite-f2-sequencing) below for further details).</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-F0-coverage" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="plots/coverage/F0_coverage.png" class="img-fluid figure-img" width="2000"></p>
<p></p><figcaption class="figure-caption">Figure&nbsp;3.1: Mean sequencing depth per chromosome for <em>Cab</em> and <em>Kaga</em> F<sub>0</sub> strains, with genome-wide mean depth across all chromosomes shown under the subtitles.</figcaption><p></p>
</figure>
</div>
</div>
</div>
</section>
<section id="f0-homozygosity" class="level2" data-number="3.2">
<h2 data-number="3.2" class="anchored" data-anchor-id="f0-homozygosity"><span class="header-section-number">3.2</span> F<sub>0</sub> homozygosity</h2>
<p>Before proceeding to map the F<sub>2</sub> sequences to the genotypes of the F<sub>0</sub> generation, I first investigated the levels of homozygosity in the F<sub>0</sub> <em>Cab</em> and <em>Kaga</em> strains, as this would affect our ability to accurately call the F<sub>2</sub> generation. That is to say, for regions where either F<sub>0</sub> parent is consistently heterozygous, it would be difficult to determine the parent from which a particular F<sub>2</sub> individual derived its chromosomes at that locus. I therefore aligned the high-coverage sequencing data for the F<sub>0</sub> <em>Cab</em> and <em>Kaga</em> strains to the medaka <em>HdrR</em> reference (Ensembl release 104, build ASM223467v1) using <em>BWA-MEM2</em> <span class="citation" data-cites="vasimuddinEfficientArchitectureawareAcceleration2019">(<a href="07-references.html#ref-vasimuddinEfficientArchitectureawareAcceleration2019" role="doc-biblioref">Vasimuddin et al. 2019</a>)</span>, sorted the aligned .sam files, marked duplicate reads, and merged the paired reads with picard <span class="citation" data-cites="Picard2019toolkit">(<a href="07-references.html#ref-Picard2019toolkit" role="doc-biblioref"><span>“Picard Toolkit”</span> 2019</a>)</span>, and indexed the .bam files with <em>SAMtools</em> <span class="citation" data-cites="liSequenceAlignmentMap2009">(<a href="07-references.html#ref-liSequenceAlignmentMap2009" role="doc-biblioref">Li et al. 2009</a>)</span>.</p>
<p>To call variants, I followed the GATK best practices (to the extent they were applicable) <span class="citation" data-cites="mckennaGenomeAnalysisToolkit2010 depristoFrameworkVariationDiscovery2011 vanderauweraGenomicsCloudUsing2020">(<a href="07-references.html#ref-mckennaGenomeAnalysisToolkit2010" role="doc-biblioref">McKenna et al. 2010</a>; <a href="07-references.html#ref-depristoFrameworkVariationDiscovery2011" role="doc-biblioref">DePristo et al. 2011</a>; <a href="07-references.html#ref-vanderauweraGenomicsCloudUsing2020" role="doc-biblioref">Van der Auwera and O’Connor 2020</a>)</span> with GATK’s HaplotypeCaller and GenotypeGVCFs tools <span class="citation" data-cites="poplinScalingAccurateGenetic2018">(<a href="07-references.html#ref-poplinScalingAccurateGenetic2018" role="doc-biblioref">Poplin et al. 2018</a>)</span>, then merged all calls into a single .vcf file with picard <span class="citation" data-cites="Picard2019toolkit">(<a href="07-references.html#ref-Picard2019toolkit" role="doc-biblioref"><span>“Picard Toolkit”</span> 2019</a>)</span>. Finally, I extracted the biallelic calls for <em>Cab</em> and <em>Kaga</em> with bcftools <span class="citation" data-cites="danecekTwelveYearsSAMtools2021">(<a href="07-references.html#ref-danecekTwelveYearsSAMtools2021" role="doc-biblioref">Danecek et al. 2021</a>)</span>, counted the number of SNPs within non-overlapping, 5-kb bins, and calculated the proportion of SNPs within each bin that were homozgyous.</p>
<p><strong>Figure @ref(fig:somite-f0-cab)</strong> is a circos plot generated with circlize <span class="citation" data-cites="guCirclizeImplementsEnhances2014">(<a href="07-references.html#ref-guCirclizeImplementsEnhances2014" role="doc-biblioref">Gu et al. 2014</a>)</span> for the <em>Cab</em> F<sub>0</sub> strain used in this experiment, featuring the proportion of homozygous SNPs per 5-kb bin (green), and the total number of SNPs in each bin (yellow). As expected for a strain that has been inbred for over 10 generations, the mean homozygosity for this strain is high, with a mean proportion of homozygosity across all bins of 83%.</p>
<div class="cell">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="plots/circos/trio_homo/hdrr/5000/Cab.png" class="img-fluid figure-img" width="1968"></p>
<p></p><figcaption class="figure-caption">Proportion of homozygous SNPs within 5 kb bins in the <em>Cab</em> F<sub>0</sub> generation genome (<em>green</em>), and number of SNPs in each bin (<em>yellow</em>).</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>However, the levels of homozygosity in the <em>Kaga</em> strain used in this experiment was far lower, with a mean homozygosity across all bins of only 31% (<strong><a href="#fig-somite-f0-kaga">Figure&nbsp;<span>3.2</span></a></strong>). This was a surprise, as it is an established strain that has been extant for decades, and we therefore expected the level of homozygosity to be commensurate with that observed in the <em>Cab</em> strain. An obvious exception is chr22, for which <em>Kaga</em> appears to be homozygous across its entire length.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-somite-f0-kaga" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="plots/circos/trio_homo/hdrr/5000/Kaga.png" class="img-fluid figure-img" width="1968"></p>
<p></p><figcaption class="figure-caption">Figure&nbsp;3.2: Proportion of homozygous SNPs within 5 kb bins in the <em>Kaga</em> F<sub>0</sub> generation genome (<em>red</em>), and number of SNPs in each bin (<em>yellow</em>).</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>To determine whether the low levels of observed homozygosity in Kaga was affected by its alignments to the southern Japanese <em>HdrR</em> reference, we also aligned the F<sub>0</sub> samples to the northern Japanese <em>HNI</em> reference (<strong><a href="#fig-somite-f0-kaga-hni">Figure&nbsp;<span>3.3</span></a></strong>). This did not make differences to the levels of observed homozygosity in either sample, which gave us confidence that the low homozygosity observed in <em>Kaga</em> was not driven by reference bias. I understand from our collaborators that the low homozygosity of this <em>Kaga</em> individual must have resulted from the strain having been contaminated at some stage by breeding with a different inbred strain prior to when they received the individuals.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-somite-f0-kaga-hni" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="plots/circos/trio_homo/hni/5000/Kaga.png" class="img-fluid figure-img" style="width:100.0%"></p>
<p></p><figcaption class="figure-caption">Figure&nbsp;3.3: Proportion of homozygous SNPs within 5 kb bins in the <em>Kaga</em> F<sub>0</sub> generation genome when aligned to the <em>HNI</em> reference (<em>red</em>), and number of SNPs in each bin (<em>yellow</em>).</figcaption><p></p>
</figure>
</div>
</div>
</div>


<div id="refs" class="references csl-bib-body hanging-indent" role="doc-bibliography" style="display: none">
<div id="ref-danecekTwelveYearsSAMtools2021" class="csl-entry" role="doc-biblioentry">
Danecek, Petr, James K Bonfield, Jennifer Liddle, John Marshall, Valeriu Ohan, Martin O Pollard, Andrew Whitwham, et al. 2021. <span>“Twelve Years of <span>SAMtools</span> and <span>BCFtools</span>.”</span> <em>GigaScience</em> 10 (2): giab008. <a href="https://doi.org/10.1093/gigascience/giab008">https://doi.org/10.1093/gigascience/giab008</a>.
</div>
<div id="ref-depristoFrameworkVariationDiscovery2011" class="csl-entry" role="doc-biblioentry">
DePristo, Mark A., Eric Banks, Ryan Poplin, Kiran V. Garimella, Jared R. Maguire, Christopher Hartl, Anthony A. Philippakis, et al. 2011. <span>“A Framework for Variation Discovery and Genotyping Using Next-Generation <span>DNA</span> Sequencing Data.”</span> <em>Nature Genetics</em> 43 (5, 5): 491–98. <a href="https://doi.org/10.1038/ng.806">https://doi.org/10.1038/ng.806</a>.
</div>
<div id="ref-guCirclizeImplementsEnhances2014" class="csl-entry" role="doc-biblioentry">
Gu, Zuguang, Lei Gu, Roland Eils, Matthias Schlesner, and Benedikt Brors. 2014. <span>“Circlize Implements and Enhances Circular Visualization in <span>R</span>.”</span> <em>Bioinformatics</em> 30 (19): 2811–12. <a href="https://doi.org/10.1093/bioinformatics/btu393">https://doi.org/10.1093/bioinformatics/btu393</a>.
</div>
<div id="ref-liSequenceAlignmentMap2009" class="csl-entry" role="doc-biblioentry">
Li, Heng, Bob Handsaker, Alec Wysoker, Tim Fennell, Jue Ruan, Nils Homer, Gabor Marth, Goncalo Abecasis, Richard Durbin, and 1000 Genome Project Data Processing Subgroup. 2009. <span>“The <span>Sequence</span> Alignment/Map (<span>SAM</span>) Format and <span>SAMtools</span>.”</span> <em>Bioinformatics</em> 25 (16): 2078–79.
</div>
<div id="ref-mckennaGenomeAnalysisToolkit2010" class="csl-entry" role="doc-biblioentry">
McKenna, Aaron, Matthew Hanna, Eric Banks, Andrey Sivachenko, Kristian Cibulskis, Andrew Kernytsky, Kiran Garimella, et al. 2010. <span>“The <span>Genome Analysis Toolkit</span>: <span>A MapReduce</span> Framework for Analyzing Next-Generation <span>DNA</span> Sequencing Data.”</span> <em>Genome Research</em> 20 (9): 1297–1303. <a href="https://doi.org/10.1101/gr.107524.110">https://doi.org/10.1101/gr.107524.110</a>.
</div>
<div id="ref-Picard2019toolkit" class="csl-entry" role="doc-biblioentry">
<span>“Picard Toolkit.”</span> 2019. <em>Broad Institute, GitHub Repository</em>. <a href="https://broadinstitute.github.io/picard/" class="uri">https://broadinstitute.github.io/picard/</a>; Broad Institute.
</div>
<div id="ref-poplinScalingAccurateGenetic2018" class="csl-entry" role="doc-biblioentry">
Poplin, Ryan, Valentin Ruano-Rubio, Mark A. DePristo, Tim J. Fennell, Mauricio O. Carneiro, Geraldine A. Van der Auwera, David E. Kling, et al. 2018. <span>“Scaling Accurate Genetic Variant Discovery to Tens of Thousands of Samples.”</span> <span>bioRxiv</span>. <a href="https://doi.org/10.1101/201178">https://doi.org/10.1101/201178</a>.
</div>
<div id="ref-vanderauweraGenomicsCloudUsing2020" class="csl-entry" role="doc-biblioentry">
Van der Auwera, Geraldine A., and Brian D. O’Connor. 2020. <em>Genomics in the Cloud: Using <span>Docker</span>, <span>GATK</span>, and <span>WDL</span> in <span>Terra</span></em>. <span>O’Reilly Media</span>.
</div>
<div id="ref-vasimuddinEfficientArchitectureawareAcceleration2019" class="csl-entry" role="doc-biblioentry">
Vasimuddin, Md, Sanchit Misra, Heng Li, and Srinivas Aluru. 2019. <span>“Efficient Architecture-Aware Acceleration of <span>BWA-MEM</span> for Multicore Systems.”</span> In <em>2019 <span>IEEE International Parallel</span> and <span>Distributed Processing Symposium</span> (<span>IPDPS</span>)</em>, 314–24. <span>IEEE</span>.
</div>
</div>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    setTimeout(function() {
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const cites = ref.parentNode.getAttribute('data-cites').split(' ');
    tippyHover(ref, function() {
      var popup = window.document.createElement('div');
      cites.forEach(function(cite) {
        var citeDiv = window.document.createElement('div');
        citeDiv.classList.add('hanging-indent');
        citeDiv.classList.add('csl-entry');
        var biblioDiv = window.document.getElementById('ref-' + cite);
        if (biblioDiv) {
          citeDiv.innerHTML = biblioDiv.innerHTML;
        }
        popup.appendChild(citeDiv);
      });
      return popup.innerHTML;
    });
  }
});
</script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="./02-phenotypes.html" class="pagination-link">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Phenotypes of interest</span></span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="./04-F1-heterozygosity.html" class="pagination-link">
        <span class="nav-page-text"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">F<sub>1</sub> heterozygosity</span></span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav>
</div> <!-- /content -->



</body></html>