<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.2.269">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Somite periodicity notebook - 6&nbsp; GWAS</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<link href="./07-references.html" rel="next">
<link href="./05-F2-genotyping.html" rel="prev">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>

  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

</head>

<body class="nav-sidebar floating">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
    <div class="container-fluid d-flex justify-content-between">
      <h1 class="quarto-secondary-nav-title"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">GWAS</span></h1>
      <button type="button" class="quarto-btn-toggle btn" aria-label="Show secondary navigation">
        <i class="bi bi-chevron-right"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="./">Somite periodicity notebook</a> 
        <div class="sidebar-tools-main">
    <a href="https://github.com/brettellebi/somites/tree/master/book/" title="Source Code" class="sidebar-tool px-1"><i class="bi bi-github"></i></a>
</div>
    </div>
      </div>
      <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
      </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">About</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./01-background.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Background</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./02-phenotypes.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Phenotypes of interest</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./03-F0-homozygosity.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">F<sub>0</sub> homozygosity</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./04-F1-heterozygosity.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">F<sub>1</sub> heterozygosity</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./05-F2-genotyping.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">F<sub>2</sub> genotyping</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./06-GWAS.html" class="sidebar-item-text sidebar-link active"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">GWAS</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./07-references.html" class="sidebar-item-text sidebar-link">References</a>
  </div>
</li>
    </ul>
    </div>
</nav>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#period-intercept" id="toc-period-intercept" class="nav-link active" data-scroll-target="#period-intercept"><span class="toc-section-number">6.1</span>  Period intercept</a></li>
  <li><a href="#psm-area" id="toc-psm-area" class="nav-link" data-scroll-target="#psm-area"><span class="toc-section-number">6.2</span>  PSM area</a></li>
  </ul>
<div class="toc-actions"><div><i class="bi bi-github"></i></div><div class="action-links"><p><a href="https://github.com/brettellebi/somites/edit/master/book/06-GWAS.qmd" class="toc-action">Edit this page</a></p></div></div></nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title"><span id="sec-somite-assoc-tests" class="quarto-section-identifier d-none d-lg-block"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">GWAS</span></span></h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  

</header>

<p>Finally, I used the called recombination blocks as pseudo-SNPs in a genetic linkage analysis. To detect associations between the pseudo-SNPs and the three phenotypes of interest, I used a linear mixed model (<strong>LMM</strong>) as implemented in GCTA <span class="citation" data-cites="yangGCTAToolGenomewide2011">(<a href="07-references.html#ref-yangGCTAToolGenomewide2011" role="doc-biblioref">Yang et al. 2011</a>)</span>, which follows the form set out in the Introduction in section @ref(GWAS), equations @ref(eq:gwaslmm) to @ref(eq:gwaslmme). For the genetic relationship matrix (<strong>GRM</strong>), I additionally used the leave-one-chromosome-out implementation of GCTA’s LMM, with excludes the chromosome on which the candidate SNP is located when calculating the GRM. A GRM constructed from the entire genome is presented in <strong><a href="#fig-somite-grm">Figure&nbsp;<span>6.1</span></a></strong>. The square in the top right-hand corner is created by samples ~550-648, which, based on Figure @ref(fig:karyo-no-missing) above, clearly have distinct genotypes to the rest of the samples due to their having been bred from different F<sub>1</sub> parents.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-somite-grm" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="plots/grm_man/hdrr/None/5000/grm_man_0.8.png" class="img-fluid figure-img" width="6000"></p>
<p></p><figcaption class="figure-caption">Figure&nbsp;6.1: GRM generated for 600 F<sub>2</sub> samples using Equations @ref(eq:grmstd) and @ref(eq:grm).</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>As described above in <strong><a href="02-phenotypes.html"><span>Chapter&nbsp;2</span></a></strong>, the microscope used to image the embryos (either AU or DB) differed by several degrees in heat, which likely caused differences in the measurements observed. I accordingly experimented with including microscope as a covariate, either alone or together with the genotype for the reporter locus (either homozygous or heterozygous), or excluding it altogether. In an attempt to avoid complications resulting from its inclusion, I also tried inverse-normalising the period phenotype within each microscope group, transforming the phenotype to fit a normal distribution across both microscopes. I onlyshow the results from this transformed phenotype in the following section.</p>
<p>To set the significance threshold, I permuted the phenotype across samples using 10 different random seeds, together with all covariates when included, and ran a separate linkage model for each permutation. I then set the lowest <span class="math inline">\(p\)</span>-value from all 10 permutation as the significance threshold for the non-permuted model. I additionally applied a Bonferroni correction to our <span class="math inline">\(p\)</span>-values by dividing <span class="math inline">\(\alpha\)</span> (0.05) by the number of pseudo-SNPs in the model, and set this as a secondary threshold.</p>
<section id="period-intercept" class="level2" data-number="6.1">
<h2 data-number="6.1" class="anchored" data-anchor-id="period-intercept"><span class="header-section-number">6.1</span> Period intercept</h2>
<p><strong><a href="#fig-somite-manhattan">Figure&nbsp;<span>6.2</span></a></strong> is a Manhattan plot of the genetic linkage results for the period intercept phenotype, inverse-normalised across microscopes. The regions found to be significant based on the permutations’ minimum <span class="math inline">\(p\)</span>-value are set out in <strong><a href="#tbl-somite-sig-int">Table&nbsp;<span>6.1</span></a></strong>.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-somite-manhattan" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="plots/gcta_invnorm/hdrr/None/5000/0.8/intercept.png" class="img-fluid figure-img" width="1920"></p>
<p></p><figcaption class="figure-caption">Figure&nbsp;6.2: Manhattan plot of the genetic linkage results for the period intercept phenotype, inverse-normalised across microscopes. Pseudo-SNPs with <span class="math inline">\(p\)</span>-values lower than the permutation significance threshold are highlighted in red.</figcaption><p></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<div class="cell-output-display">
<div id="tbl-somite-sig-int" class="anchored">
<table class="table table-sm table-striped">
<caption>Table&nbsp;6.1: Significant 5-kb bin ranges for period intercept below the minimum p-value from 10 permutations.</caption>
<thead>
<tr class="header">
<th style="text-align: right;">Chromosome</th>
<th style="text-align: right;">Bin start</th>
<th style="text-align: right;">Bin end</th>
<th style="text-align: right;">Length (kb)</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: right;">3</td>
<td style="text-align: right;">31880001</td>
<td style="text-align: right;">35420000</td>
<td style="text-align: right;">3540</td>
</tr>
<tr class="even">
<td style="text-align: right;">4</td>
<td style="text-align: right;">18090001</td>
<td style="text-align: right;">18095000</td>
<td style="text-align: right;">5</td>
</tr>
<tr class="odd">
<td style="text-align: right;">10</td>
<td style="text-align: right;">2995001</td>
<td style="text-align: right;">3690000</td>
<td style="text-align: right;">695</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<p>These regions contained a total of 46,872 SNPs imputed from the genotype of the F<sub>0</sub> parental strains. We ran Ensembl’s Variant Effect Predictor <span class="citation" data-cites="mclarenEnsemblVariantEffect2016">(<a href="07-references.html#ref-mclarenEnsemblVariantEffect2016" role="doc-biblioref">McLaren et al. 2016</a>)</span> over these SNPs to identify those that would be most likely to have functional consequences. The full counts of SNPs falling into each category of ‘consequence’ are set out in <strong><a href="#tbl-int-consequence">Table&nbsp;<span>6.2</span></a></strong>. From this process I identified 38 genes that included a missense variant, 1 that included a missense variant and a start lost (ENSORLG00000014616), and 1 that included a missense variant and a stop lost (ENSORLG00000015149).</p>
<div class="cell">
<div class="cell-output-display">
<div id="tbl-int-consequence" class="anchored">
<table class="table table-sm table-striped">
<caption>Table&nbsp;6.2: Variant Effect Predictor results for SNPs in the bins.</caption>
<thead>
<tr class="header">
<th style="text-align: left;">Consequence</th>
<th style="text-align: right;">Count</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">intron variant</td>
<td style="text-align: right;">47211</td>
</tr>
<tr class="even">
<td style="text-align: left;">intergenic variant</td>
<td style="text-align: right;">20045</td>
</tr>
<tr class="odd">
<td style="text-align: left;">upstream gene variant</td>
<td style="text-align: right;">7304</td>
</tr>
<tr class="even">
<td style="text-align: left;">downstream gene variant</td>
<td style="text-align: right;">5229</td>
</tr>
<tr class="odd">
<td style="text-align: left;">3 prime UTR variant</td>
<td style="text-align: right;">1082</td>
</tr>
<tr class="even">
<td style="text-align: left;">synonymous variant</td>
<td style="text-align: right;">694</td>
</tr>
<tr class="odd">
<td style="text-align: left;">missense variant</td>
<td style="text-align: right;">383</td>
</tr>
<tr class="even">
<td style="text-align: left;">5 prime UTR variant</td>
<td style="text-align: right;">201</td>
</tr>
<tr class="odd">
<td style="text-align: left;">splice region variant,intron variant</td>
<td style="text-align: right;">126</td>
</tr>
<tr class="even">
<td style="text-align: left;">missense variant,splice region variant</td>
<td style="text-align: right;">19</td>
</tr>
<tr class="odd">
<td style="text-align: left;">splice region variant,synonymous variant</td>
<td style="text-align: right;">17</td>
</tr>
<tr class="even">
<td style="text-align: left;">stop gained</td>
<td style="text-align: right;">3</td>
</tr>
<tr class="odd">
<td style="text-align: left;">splice donor variant</td>
<td style="text-align: right;">1</td>
</tr>
<tr class="even">
<td style="text-align: left;">start lost</td>
<td style="text-align: right;">1</td>
</tr>
<tr class="odd">
<td style="text-align: left;">stop lost</td>
<td style="text-align: right;">1</td>
</tr>
<tr class="even">
<td style="text-align: left;">stop lost,splice region variant</td>
<td style="text-align: right;">1</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<p>My collaborators then combined these results with bulk RNA-seq that they had performed on F<sub>0</sub> <em><span style="     color: #43AA8B !important;">Cab</span></em> and <em><span style="     color: #DE3C4B !important;">Kaga</span></em> individuals, to determine which of these genes are expressed in the tail during embryogenesis. This allowed them to reduce to the list to 29 genes, and a gene ontology analysis of this found that the list of genes was enriched for body axis, somitogenesis, and segmentation (<strong><a href="#tbl-psm-final-genes">Table&nbsp;<span>6.3</span></a></strong>). For this list of genes, our collaborators are now in the process of knocking out the protein-altering <em><span style="     color: #43AA8B !important;">Cab</span></em> allele into <em><span style="     color: #DE3C4B !important;">Kaga</span></em> embryos, and <em>vice versa</em>, to functionally validate these variants.</p>
<div class="cell">
<div class="cell-output-display">
<div id="tbl-psm-final-genes" class="anchored">

<table><caption>Table&nbsp;6.3:  Target genes for functional validation expressed in the unsegmented PSM and containing protein alterations. Table generated by Ali Seleit. </caption>
 <thead>
  <tr>
   <th style="text-align:right;"> Chromosome </th>
   <th style="text-align:left;"> Ensembl gene ID </th>
   <th style="text-align:left;"> Description </th>
   <th style="text-align:left;"> Role </th>
  </tr>
 </thead>
<tbody>
  <tr>
   <td style="text-align:right;"> 3 </td>
   <td style="text-align:left;"> ENSORLG00000014656 </td>
   <td style="text-align:left;"> mesoderm posterior protein 2-like </td>
   <td style="text-align:left;"> Somitogenesis </td>
  </tr>
  <tr>
   <td style="text-align:right;"> 3 </td>
   <td style="text-align:left;"> ENSORLG00000014659 </td>
   <td style="text-align:left;"> mesp </td>
   <td style="text-align:left;"> Somitogenesis </td>
  </tr>
  <tr>
   <td style="text-align:right;"> 10 </td>
   <td style="text-align:left;"> ENSORLG00000020474 </td>
   <td style="text-align:left;"> protocadherin 10b </td>
   <td style="text-align:left;"> Somitogenesis </td>
  </tr>
  <tr>
   <td style="text-align:right;"> 3 </td>
   <td style="text-align:left;"> ENSORLG00000014616 </td>
   <td style="text-align:left;"> NA </td>
   <td style="text-align:left;"> Possible role in Somitogenesis (MAP-kinase) </td>
  </tr>
  <tr>
   <td style="text-align:right;"> 10 </td>
   <td style="text-align:left;"> ENSORLG00000020551 </td>
   <td style="text-align:left;"> FAT atypical cadherin 4 </td>
   <td style="text-align:left;"> Possible role in Somitogenesis (PCP, Yap1 regulator) </td>
  </tr>
  <tr>
   <td style="text-align:right;"> 10 </td>
   <td style="text-align:left;"> ENSORLG00000020531 </td>
   <td style="text-align:left;"> neurogenin 1 </td>
   <td style="text-align:left;"> Possible role in Somitogenesis (BHLH-TF regulation of Wnt) </td>
  </tr>
  <tr>
   <td style="text-align:right;"> 3 </td>
   <td style="text-align:left;"> ENSORLG00000015149 </td>
   <td style="text-align:left;"> ADAM metallopeptidase with thrombospondin type 1 motif 18 </td>
   <td style="text-align:left;"> Extracellular matrix </td>
  </tr>
  <tr>
   <td style="text-align:right;"> 3 </td>
   <td style="text-align:left;"> ENSORLG00000015418 </td>
   <td style="text-align:left;"> matrix metallopeptidase 15 </td>
   <td style="text-align:left;"> Extracellular matrix </td>
  </tr>
  <tr>
   <td style="text-align:right;"> 10 </td>
   <td style="text-align:left;"> ENSORLG00000020488 </td>
   <td style="text-align:left;"> transforming growth factor beta induced </td>
   <td style="text-align:left;"> Extracellular matrix </td>
  </tr>
  <tr>
   <td style="text-align:right;"> 3 </td>
   <td style="text-align:left;"> ENSORLG00000028055 </td>
   <td style="text-align:left;"> NA </td>
   <td style="text-align:left;"> &nbsp;Signal trasduction (Rho) </td>
  </tr>
  <tr>
   <td style="text-align:right;"> 10 </td>
   <td style="text-align:left;"> ENSORLG00000020481 </td>
   <td style="text-align:left;"> ArfGAP with RhoGAP domain, ankyrin repeat and PH domain 3 </td>
   <td style="text-align:left;"> &nbsp;Signal transduction (GTPase activator RhoGap) </td>
  </tr>
  <tr>
   <td style="text-align:right;"> 10 </td>
   <td style="text-align:left;"> ENSORLG00000020525 </td>
   <td style="text-align:left;"> TBC1 domain family member 2A-like </td>
   <td style="text-align:left;"> Signal transduction Gtpase activator, cadherin recycling </td>
  </tr>
  <tr>
   <td style="text-align:right;"> 3 </td>
   <td style="text-align:left;"> ENSORLG00000015260 </td>
   <td style="text-align:left;"> synapse associated protein 1 </td>
   <td style="text-align:left;"> TOR acitivity </td>
  </tr>
  <tr>
   <td style="text-align:right;"> 10 </td>
   <td style="text-align:left;"> ENSORLG00000028553 </td>
   <td style="text-align:left;"> NA </td>
   <td style="text-align:left;"> NA </td>
  </tr>
  <tr>
   <td style="text-align:right;"> 3 </td>
   <td style="text-align:left;"> ENSORLG00000015460 </td>
   <td style="text-align:left;"> dpy-19 like C-mannosyltransferase 3 </td>
   <td style="text-align:left;"> glycosylation </td>
  </tr>
  <tr>
   <td style="text-align:right;"> 10 </td>
   <td style="text-align:left;"> ENSORLG00000022010 </td>
   <td style="text-align:left;"> beta-1,4-galactosyltransferase 1 </td>
   <td style="text-align:left;"> glycosylation </td>
  </tr>
  <tr>
   <td style="text-align:right;"> 10 </td>
   <td style="text-align:left;"> ENSORLG00000020498 </td>
   <td style="text-align:left;"> protein NipSnap homolog 3A </td>
   <td style="text-align:left;"> Mitochondria </td>
  </tr>
  <tr>
   <td style="text-align:right;"> 10 </td>
   <td style="text-align:left;"> ENSORLG00000020494 </td>
   <td style="text-align:left;"> nitric oxide associated 1 </td>
   <td style="text-align:left;"> Mitochondria </td>
  </tr>
  <tr>
   <td style="text-align:right;"> 10 </td>
   <td style="text-align:left;"> ENSORLG00000020504 </td>
   <td style="text-align:left;"> ATP-binding cassette sub-family A member 1 </td>
   <td style="text-align:left;"> Metabolism (cholestrol efflux) </td>
  </tr>
  <tr>
   <td style="text-align:right;"> 3 </td>
   <td style="text-align:left;"> ENSORLG00000015118 </td>
   <td style="text-align:left;"> phosphorylase kinase regulatory subunit beta </td>
   <td style="text-align:left;"> Metabolism </td>
  </tr>
  <tr>
   <td style="text-align:right;"> 10 </td>
   <td style="text-align:left;"> ENSORLG00000020493 </td>
   <td style="text-align:left;"> RE1 silencing transcription factor </td>
   <td style="text-align:left;"> RNA pol II (transcritpion regulation </td>
  </tr>
  <tr>
   <td style="text-align:right;"> 3 </td>
   <td style="text-align:left;"> ENSORLG00000015365 </td>
   <td style="text-align:left;"> solute carrier family 7 member 6 opposite strand </td>
   <td style="text-align:left;"> RNA pol II (nuclear export) </td>
  </tr>
  <tr>
   <td style="text-align:right;"> 10 </td>
   <td style="text-align:left;"> ENSORLG00000029052 </td>
   <td style="text-align:left;"> exosome component 3 </td>
   <td style="text-align:left;"> rRNA + RNA binding </td>
  </tr>
  <tr>
   <td style="text-align:right;"> 3 </td>
   <td style="text-align:left;"> ENSORLG00000015278 </td>
   <td style="text-align:left;"> adhesion G protein-coupled receptor G3 </td>
   <td style="text-align:left;"> G-protein coupled receptor </td>
  </tr>
  <tr>
   <td style="text-align:right;"> 3 </td>
   <td style="text-align:left;"> ENSORLG00000015287 </td>
   <td style="text-align:left;"> adhesion G-protein coupled receptor G5 </td>
   <td style="text-align:left;"> G-protein coupled receptor </td>
  </tr>
  <tr>
   <td style="text-align:right;"> 10 </td>
   <td style="text-align:left;"> ENSORLG00000025674 </td>
   <td style="text-align:left;"> matrin-3 </td>
   <td style="text-align:left;"> inner nuclear protein (chromatin architecture </td>
  </tr>
  <tr>
   <td style="text-align:right;"> 10 </td>
   <td style="text-align:left;"> ENSORLG00000023325 </td>
   <td style="text-align:left;"> matrin-3 </td>
   <td style="text-align:left;"> inner nuclear protein (chromatin architecture </td>
  </tr>
  <tr>
   <td style="text-align:right;"> 10 </td>
   <td style="text-align:left;"> ENSORLG00000022388 </td>
   <td style="text-align:left;"> poly(A) binding protein interacting protein 2 </td>
   <td style="text-align:left;"> Translation repressor </td>
  </tr>
  <tr>
   <td style="text-align:right;"> 3 </td>
   <td style="text-align:left;"> ENSORLG00000015096 </td>
   <td style="text-align:left;"> integrin alpha FG-GAP repeat containing 1 </td>
   <td style="text-align:left;"> NA possibly T cell activation </td>
  </tr>
</tbody>
</table>

</div>
</div>
</div>
<p>My collaborators have begun the process of using the CRISPR-Cas9 system <span class="citation" data-cites="campenhoutGuidelinesOptimizedGene2019">(<a href="07-references.html#ref-campenhoutGuidelinesOptimizedGene2019" role="doc-biblioref">Campenhout et al. 2019</a>)</span> to knock out some of these candidate genes to determine their effect on the period phenotype. The <em>mespb</em> (ENSORLG00000014656) and <em>pcdh10b</em> (ENSORLG00000020474) genes have both been found to be involved in segmental boundary formation and somite patterning <span class="citation" data-cites="hitachiPhysicalInteractionTbx62008 rangarajanPCNSNovelProtocadherin2006">(<a href="07-references.html#ref-hitachiPhysicalInteractionTbx62008" role="doc-biblioref">Hitachi et al. 2008</a>; <a href="07-references.html#ref-rangarajanPCNSNovelProtocadherin2006" role="doc-biblioref">Rangarajan, Luo, and Sargent 2006</a>)</span>, and are highly conserved in vertebrates <span class="citation" data-cites="gulEvolutionDiversityCadherins2017 satouAscidianMespGene2004">(<a href="07-references.html#ref-gulEvolutionDiversityCadherins2017" role="doc-biblioref">Gul et al. 2017</a>; <a href="07-references.html#ref-satouAscidianMespGene2004" role="doc-biblioref">Satou, Imai, and Satoh 2004</a>)</span>. Preliminary results have shown that knocking out either of these genes from the <em><span style="     color: #43AA8B !important;">Cab</span></em> strain significantly speeds up somite period development by 2-3 minutes (<strong><a href="#fig-ko-period">Figure&nbsp;<span>6.3</span></a>)</strong>), providing promising evidence that these genes are involved in the establishment of the segmentation clock.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-ko-period" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="plots/ali/ko_results.png" class="img-fluid figure-img" width="2000"></p>
<p></p><figcaption class="figure-caption">Figure&nbsp;6.3: Period intercept for <em>Cab</em> embryos with CRISPR-Cas9 knock-outs <span class="citation" data-cites="campenhoutGuidelinesOptimizedGene2019">(<a href="07-references.html#ref-campenhoutGuidelinesOptimizedGene2019" role="doc-biblioref">Campenhout et al. 2019</a>)</span> of <em>mespb</em> and <em>pcdh10b</em> compared to controls. Figure generated by Ali Seleit.</figcaption><p></p>
</figure>
</div>
</div>
</div>
</section>
<section id="psm-area" class="level2" data-number="6.2">
<h2 data-number="6.2" class="anchored" data-anchor-id="psm-area"><span class="header-section-number">6.2</span> PSM area</h2>
<p><strong><a href="#fig-psm-manhattan">Figure&nbsp;<span>6.4</span></a></strong> is a Manhattan plot of the genetic linkage results for the PSM area phenotype. The regions found to be significant based on the permutations’ minimum <span class="math inline">\(p\)</span>-value are set out in <strong><a href="#tbl-somite-sig-psm">Table&nbsp;<span>6.4</span></a></strong>, although they exceed the Bonferroni correction threshold as well. I note that this ~6 Mb significant region on chromosome 3 does not overlap at all with the significant region discovered for the period intercept phenotype.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-psm-manhattan" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="plots/gcta/hdrr/None/5000/0.8/unsegmented_psm_area/None.png" class="img-fluid figure-img" width="1920"></p>
<p></p><figcaption class="figure-caption">Figure&nbsp;6.4: Manhattan plot of the genetic linkage results for the PSM area phenotype. Pseudo-SNPs with <span class="math inline">\(p\)</span>-values lower than the permutation significance threshold are highlighted in yellow.</figcaption><p></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<div class="cell-output-display">
<div id="tbl-somite-sig-psm" class="anchored">

<table><caption>Table&nbsp;6.4:  Significant 5-kb bin range for PSM area below the minimum p-value from 10 permutations. </caption>
 <thead>
  <tr>
   <th style="text-align:right;"> Chromosome </th>
   <th style="text-align:right;"> Bin start </th>
   <th style="text-align:right;"> Bin end </th>
   <th style="text-align:right;"> Length (kb) </th>
  </tr>
 </thead>
<tbody>
  <tr>
   <td style="text-align:right;"> 3 </td>
   <td style="text-align:right;"> 20375001 </td>
   <td style="text-align:right;"> 26285000 </td>
   <td style="text-align:right;"> 5910 </td>
  </tr>
</tbody>
</table>

</div>
</div>
</div>
<p>This region contained a total of 29,096 SNPs imputed from the genotype of the F<sub>0</sub> parental strains. I ran Ensembl’s Variant Effect Predictor <span class="citation" data-cites="mclarenEnsemblVariantEffect2016">(<a href="07-references.html#ref-mclarenEnsemblVariantEffect2016" role="doc-biblioref">McLaren et al. 2016</a>)</span> over these SNPs to identify those that would be most likely to have functional consequences. The full counts of SNPs falling into each category of ‘consequence’ are set out in <strong><a href="#tbl-psm-consequence">Table&nbsp;<span>6.5</span></a></strong>. From this process I identified 114 genes that included a missense variant, and 1 that included a both a missense variant and a start lost (ENSORLG00000010863, a centriole, cilia and spindle-associated protein).</p>
<div class="cell">
<div class="cell-output-display">
<div id="tbl-psm-consequence" class="anchored">

<table><caption>Table&nbsp;6.5:  Variant Effect Predictor results for SNPs in the bins. </caption>
 <thead>
  <tr>
   <th style="text-align:left;"> Consequence </th>
   <th style="text-align:right;"> Count </th>
  </tr>
 </thead>
<tbody>
  <tr>
   <td style="text-align:left;"> intron variant </td>
   <td style="text-align:right;"> 47211 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> intergenic variant </td>
   <td style="text-align:right;"> 20045 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> upstream gene variant </td>
   <td style="text-align:right;"> 7304 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> downstream gene variant </td>
   <td style="text-align:right;"> 5229 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 3 prime UTR variant </td>
   <td style="text-align:right;"> 1082 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> synonymous variant </td>
   <td style="text-align:right;"> 694 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> missense variant </td>
   <td style="text-align:right;"> 383 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> 5 prime UTR variant </td>
   <td style="text-align:right;"> 201 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> splice region variant,intron variant </td>
   <td style="text-align:right;"> 126 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> missense variant,splice region variant </td>
   <td style="text-align:right;"> 19 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> splice region variant,synonymous variant </td>
   <td style="text-align:right;"> 17 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> stop gained </td>
   <td style="text-align:right;"> 3 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> splice donor variant </td>
   <td style="text-align:right;"> 1 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> start lost </td>
   <td style="text-align:right;"> 1 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> stop lost </td>
   <td style="text-align:right;"> 1 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> stop lost,splice region variant </td>
   <td style="text-align:right;"> 1 </td>
  </tr>
</tbody>
</table>

</div>
</div>
</div>
<p>Our collaborators then combined these results with bulk RNA-seq that they had performed on F<sub>0</sub> <em><span style="     color: #43AA8B !important;">Cab</span></em> and <em><span style="     color: #DE3C4B !important;">Kaga</span></em> individuals, to determine which of these genes are expressed in the unsegmented tail during embryogenesis. This allowed them to reduce to the list to 96 genes, although they were not apparently associated with a specific gene ontology as perhaps expected, given this is a single locus that is potentially driven by just one gene. As with the period intercept phenotype, our collaborators are now in the process of knocking in the <em><span style="     color: #43AA8B !important;">Cab</span></em> allele into <em><span style="     color: #DE3C4B !important;">Kaga</span></em> embryos, and <em>vice versa</em>, to functionally validate these variants.</p>


<div id="refs" class="references csl-bib-body hanging-indent" role="doc-bibliography" style="display: none">
<div id="ref-campenhoutGuidelinesOptimizedGene2019" class="csl-entry" role="doc-biblioentry">
Campenhout, Claude Van, Pauline Cabochette, Anne-Clémence Veillard, Miklos Laczik, Agnieszka Zelisko-Schmidt, Céline Sabatel, Maxime Dhainaut, Benoit Vanhollebeke, Cyril Gueydan, and Véronique Kruys. 2019. <span>“Guidelines for Optimized Gene Knockout Using <span>CRISPR</span>/<span>Cas9</span>.”</span> <em>BioTechniques</em> 66 (6): 295–302. <a href="https://doi.org/10.2144/btn-2018-0187">https://doi.org/10.2144/btn-2018-0187</a>.
</div>
<div id="ref-gulEvolutionDiversityCadherins2017" class="csl-entry" role="doc-biblioentry">
Gul, Ismail Sahin, Paco Hulpiau, Yvan Saeys, and Frans van Roy. 2017. <span>“Evolution and Diversity of Cadherins and Catenins.”</span> <em>Experimental Cell Research</em>, Cell <span>Sensing</span> and <span>Signaling</span> via <span>Cell-Cell Adhesion</span>, 358 (1): 3–9. <a href="https://doi.org/10.1016/j.yexcr.2017.03.001">https://doi.org/10.1016/j.yexcr.2017.03.001</a>.
</div>
<div id="ref-hitachiPhysicalInteractionTbx62008" class="csl-entry" role="doc-biblioentry">
Hitachi, Keisuke, Hiroki Danno, Akiko Kondow, Kiyoshi Ohnuma, Hideho Uchiyama, Shoichi Ishiura, Akira Kurisaki, and Makoto Asashima. 2008. <span>“Physical Interaction Between <span>Tbx6</span> and Mespb Is Indispensable for the Activation of Bowline Expression During <span>Xenopus</span> Somitogenesis.”</span> <em>Biochemical and Biophysical Research Communications</em> 372 (4): 607–12. <a href="https://doi.org/10.1016/j.bbrc.2008.05.083">https://doi.org/10.1016/j.bbrc.2008.05.083</a>.
</div>
<div id="ref-mclarenEnsemblVariantEffect2016" class="csl-entry" role="doc-biblioentry">
McLaren, William, Laurent Gil, Sarah E. Hunt, Harpreet Singh Riat, Graham R. S. Ritchie, Anja Thormann, Paul Flicek, and Fiona Cunningham. 2016. <span>“The <span>Ensembl Variant Effect Predictor</span>.”</span> <em>Genome Biology</em> 17 (1): 122. <a href="https://doi.org/10.1186/s13059-016-0974-4">https://doi.org/10.1186/s13059-016-0974-4</a>.
</div>
<div id="ref-rangarajanPCNSNovelProtocadherin2006" class="csl-entry" role="doc-biblioentry">
Rangarajan, Janaki, Ting Luo, and Thomas D. Sargent. 2006. <span>“<span>PCNS</span>: <span>A</span> Novel Protocadherin Required for Cranial Neural Crest Migration and Somite Morphogenesis in <span>Xenopus</span>.”</span> <em>Developmental Biology</em> 295 (1): 206–18. <a href="https://doi.org/10.1016/j.ydbio.2006.03.025">https://doi.org/10.1016/j.ydbio.2006.03.025</a>.
</div>
<div id="ref-satouAscidianMespGene2004" class="csl-entry" role="doc-biblioentry">
Satou, Yutaka, Kaoru S. Imai, and Nori Satoh. 2004. <span>“The Ascidian <span>Mesp</span> Gene Specifies Heart Precursor Cells.”</span> <em>Development</em> 131 (11): 2533–41. <a href="https://doi.org/10.1242/dev.01145">https://doi.org/10.1242/dev.01145</a>.
</div>
<div id="ref-yangGCTAToolGenomewide2011" class="csl-entry" role="doc-biblioentry">
Yang, Jian, S. Hong Lee, Michael E. Goddard, and Peter M. Visscher. 2011. <span>“<span>GCTA</span>: <span>A Tool</span> for <span class="nocase">Genome-wide Complex Trait Analysis</span>.”</span> <em>The American Journal of Human Genetics</em> 88 (1): 76–82. <a href="https://doi.org/10.1016/j.ajhg.2010.11.011">https://doi.org/10.1016/j.ajhg.2010.11.011</a>.
</div>
</div>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="./05-F2-genotyping.html" class="pagination-link">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">F<sub>2</sub> genotyping</span></span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="./07-references.html" class="pagination-link">
        <span class="nav-page-text">References</span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav>
</div> <!-- /content -->



</body></html>