---
title: "Meeting log"
author: "Ian Brettell"
date: '`r format(Sys.Date())`'
output:
  html_document:
    toc: true
    toc_float: true
    dev: 'svg'
    number_sections: false
    keep_md: false
    pandoc_args: --lua-filter=color-text.lua
    highlight: pygments
---

# 15 July 2021

Notes for Ali:

* Tried sample 171 again (the one you uploaded again), and it failed again at the alignment stage with the following error:

```{bash, eval = F}
[mem_sam_pe_batch_post] paired reads have different names: "�0�����O&�����x�7�Z�HĻ�@�%Q�ü��", "ST-K00119:220:HKNVLBBXY:7:1101:1661:1138"
```

* Fixed the sample names so that they don't include the pool number at the beginning, e.g. sample 171 is no longer called 7171

* Cab has 15M homozygous sites, whereas Kaga only has 5M. Reason: Cab aligns better to HdrR, so there are many more 0/0 calls?

* 3.48M homozygous sites after joining Cab and Kaga homozygous site DFs.

* 2.7M homozygous divergent sites.

# 13 July 2021

## Attendees:

- Ewan Birney
- Alexander Aulehla
- Tomas Fitzgerald
- Ali Seleit
- Ian Brettell

## Notes:

* Genetics:
  - Increase block size?
    - Try site-by-site
    - HMM would need to work on a *count* system so that it doesn't get skewed
* [**ACTION**]{color="red"} (Ian): 
  - Check colours:
    - Would expect Cab skew, especially as it has the reporter.
* [**ACTION**]{color="red"} (Ian):
  - Find number of reads covering each block for each individual.
  - If they reach < 5, increase the block size.
  - Also check out chr 16 reporter locus.
  - Make histogram of counts per block for each individual.
  - Data frame:
    - Fish  ID
    - Block ID
    - Allele 
    - Count
  - Plot a boxplot of counts per allele for all blocks per individual.
  - Do for 5, 10 and 15 kb windows, then make boxplots to compare.
  - Expect coverage across the homo-divergent regions is quite uneven.
  - Also estimate heritability? 
* [**ACTION**  ]{color="red"} (Ali):
  - Eyeball high phenotype points
  - Also look at the plate position. 
  - Is there a between-day effect? (Note: Can only image 10 individuals per day)
  - Run a Kruskal-Wallis test for:
    - day
    - microscope (if they differ)
    - any other way you can split it up
    - look at the within-individual variance too
* Re: sample size:
  - Pause imaging while we check the above.
  - Want to see the parents tight, and the F2 stretching out.
  



